### DISCUSSION TITLE
Isolation distance and Lratio

### DISCUSSION THREAD

**Discussion by jesusmartin-cortecero** [petersenpeter/CellExplorer#62] [General]
Hi Peter.

I am trying to get the quality metrics from the cell_metrics structure, but i can only find the "refractoryPeriodViolation"
should "lRatio" and "isolationDistance be in the cell_metrics structure as well?
Thanks in advance

**Comment by petersenpeter** [comment]
Hi Jesus

The L-ratio and isolation distance are not calculated by default, as they depend on the pipeline and data format. You need a low-dimensional feature space representation for this. 

The code for the metrics that takes a PCA-space input (N by D array of feature vectors (N spikes, D dimensional feature space)) is available with CellExplorer: `calc_IsolationDistance.m`and `L_ratio_calc.m`. I was planning to write a universal script for this, but have not had time to finish it. There is a script included with CellExplorer for the Neurosuite format: `LoadNeurosuiteFeatures.m`

Nick Steinmetz has written for the Kilosort/Phy format here: https://github.com/cortex-lab/sortingQuality

**Reply by jesusmartin-cortecero** [reply]
Hi Peter
thanks for the information i will try the code you suggested.
